### DISCUSSION TITLE
Issues upgrading from 2.3.130 to 2.3.140

### DISCUSSION THREAD

**Discussion by tqtran** [Security-Onion-Solutions/securityonion#8443] [Unsupported Versions]
Hello,

This is my first time posting, so please forgive me if I am not following forum etiquette.

We run a distributed on-premise Security Onion setup with a dedicated Manager, Search, and 3 Forward Nodes.  Everything has been running and updating fine for the past year.

On 7/28, I ran soup to upgrade the system from 2.3.130 to 2.3.140.  At the end of the patch, the prompt says to let the changes propagate to the nodes, so I let it sit for the entire weekend.

Today, I noticed that Kibana was showing 404 from the SOC menu, so I went ahead and restarted the Manager node.  It came up properly, but Kibana was still showing 404, and all the other nodes have disappeared from the Grid view.  So I went ahead and restarted Search, then the 3 Forward Nodes.  After about 30 minutes, nothing was coming up, so I did so-status command, all of them were showing START_WAIT.  I waited another hour, and nothing changed, so I ran so-start on each service.  Most came up, except logstash and filebeath.

This is what I got for "so-start filebeat" on a sensor (map.jinja line 7 is line 19 on this log file).  This happens on both Filebeat and Logstash on the Search node, and all 3 sensors.

      1 =========================================================================
      2 Starting filebeat...
      3
      4 This could take a while if another Salt job is running.
      5 Run this command with --force to stop all Salt jobs before proceeding.
      6 =========================================================================
      7 local:
      8     Data failed to compile:
      9 ----------
     10     Rendering SLS 'base:filebeat' failed: Jinja variable 'None' has no attribute 'items'
     11 /var/cache/salt/minion/files/base/filebeat/map.jinja(7):
     12 ---
     13 [...]
     14 {% set FILEBEAT_EXTRA_HOSTS = [] %}
     15 {% set mainint = salt['pillar.get']('host:mainint') %}
     16 {% set localhostip = salt['grains.get']('ip_interfaces').get(mainint)[0] %}
     17 {% if role in ['so-sensor', 'so-fleet', 'so-node', 'so-idh'] %}
     18   {% set node_data = salt['pillar.get']('logstash:nodes') %}
     19   {% for node_type, node_details in node_data.items() | sort %}    <======================
     20     {% if node_type in ['manager', 'managersearch', 'standalone', 'receiver' ] %}
     21       {% for hostname in node_data[node_type].keys() %}
     22         {% do FILEBEAT_EXTRA_HOSTS.append({hostname:node_details[hostname].ip}) %}
     23       {% endfor %}
     24     {% endif %}
     25 [...]
     26 ---

I tried so-elastic-restart, and it was complaining about so-filebeat and so-logstash containers do not exist.

UPDATE:  I edited the soversion file and set it back to 2.3.130 and reran soup, it resolved the so-flebeat and so-logstash, but Kibana link still shows 404.  All containers show OK with so-status on all nodes.

However, when I run curl -X GET "http://localhost:5601/app/kibana", I would get "curl: (56) Recv failure: Connection reset by peer"

I would like to avoid having to tear everything down and start over.  

Any pointers is greatly appreciated.

Thank you in advance.

**Comment by Hammer-Bear** [comment]
did you follow https://docs.securityonion.net/en/2.3/soup.html#elastic-8? if your original install was pre-2.3.60 you would need to enable so-elastic-auth prior to updating.
```
FORCE_SYNC=true so-user sync
so-kibana-restart
so-kibana-space-defaults
```
has resolved the 404 issues for some people as well

**Reply by tqtran** [reply]
Thank you for the reply.  Since I took over this project when it was already 2.3.90, I didn't know what the original install was.  I did try the enable so-elastic-auth but it told me it was already enabled.  I will try the other commands and will report back.

Digging through Kibana logs, I saw that it was complaining about saved object migration. "Unable to complete saved object migrations for the [.kibana_task_manager] index. Please check the health of your Elastic search cluster and try again. Unexpected Elasticsearch ResponseError: statusCode: 500"

**Reply by tqtran** [reply]
The commands ran fine, except for the last one

so-kibana-space-defaults shows: 

Waiting for value 'default' at 'http://localhost:5601/api/spaces/space/default' (1/300)
Server is not ready

All the way to 300

**Reply by weslambert** [reply]
Are you seeing any other errors in the `/opt/so/logs/kibana/kibana.log`?

Are there any clues in the logs in `/opt/so/log/elasticsearch/`? 

What does the cluster health look like if you run the following?

`so-elasticsearch-query _cluster/health?pretty`

**Reply by tqtran** [reply]
Thank you for the suggestions

This is what I get from the search node:
{
  "cluster_name" : "dis-so2-search01",
  "status" : "yellow",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 124,
  "active_shards" : 124,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 17,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 87.94326241134752
}

This is what I get from the manager node:
{
  "cluster_name" : "dis-so2-manager",
  "status" : "yellow",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 117,
  "active_shards" : 117,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 17,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 87.31343283582089
}

The only thing I could find in the logs that would suggest an error:

{"ecs":{"version":"8.0.0"},"@timestamp":"2022-08-16T00:07:56.827+00:00","message":"[.kibana_task_manager] Unexpected Elasticsearch ResponseError: statusCode: 500, method: POST, url: /_aliases error: [illegal_state_exception]: alias [.kibana_task_manager] has is_hidden set to true on indices [.kibana_task_manager_8.3.2_001] but does not have is_hidden set to true on indices [.kibana_task_manager_7.13.2_001]; alias must have the same is_hidden setting on all indices,","log":{"level":"ERROR","logger":"savedobjects-service"},"process":{"pid":7},"trace":{"id":"c4729231c8755337d67403e9e702f4ce"},"transaction":{"id":"50b0d300ed21358b"}}

{"ecs":{"version":"8.0.0"},"@timestamp":"2022-08-16T00:07:56.828+00:00","message":"Unable to complete saved object migrations for the [.kibana_task_manager] index. Please check the health of your Elasticsearch cluster and try again. Unexpected Elasticsearch ResponseError: statusCode: 500, method: POST, url: /_aliases error: [illegal_state_exception]: alias [.kibana_task_manager] has is_hidden set to true on indices [.kibana_task_manager_8.3.2_001] but does not have is_hidden set to true on indices [.kibana_task_manager_7.13.2_001]; alias must have the same is_hidden setting on all indices,","error":{"message":"Unable to complete saved object migrations for the [.kibana_task_manager] index. Please check the health of your Elasticsearch cluster and try again. Unexpected Elasticsearch ResponseError: statusCode: 500, method: POST, url: /_aliases error: [illegal_state_exception]: alias [.kibana_task_manager] has is_hidden set to true on indices [.kibana_task_manager_8.3.2_001] but does not have is_hidden set to true on indices [.kibana_task_manager_7.13.2_001]; alias must have the same is_hidden setting on all indices,","type":"Error","stack_trace":"Error: Unable to complete saved object migrations for the [.kibana_task_manager] index. Please check the health of your Elasticsearch cluster and try again. Unexpected Elasticsearch ResponseError: statusCode: 500, method: POST, url: /_aliases error: [illegal_state_exception]: alias [.kibana_task_manager] has is_hidden set to true on indices [.kibana_task_manager_8.3.2_001] but does not have is_hidden set to true on indices [.kibana_task_manager_7.13.2_001]; alias must have the same is_hidden setting on all indices,\n    at migrationStateActionMachine (/usr/share/kibana/src/core/server/saved_objects/migrations/migrations_state_action_machine.js:164:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Promise.all (index 1)\n    at SavedObjectsService.start (/usr/share/kibana/src/core/server/saved_objects/saved_objects_service.js:214:9)\n    at Server.start (/usr/share/kibana/src/core/server/server.js:363:31)\n    at Root.start (/usr/share/kibana/src/core/server/root/index.js:69:14)\n    at bootstrap (/usr/share/kibana/src/core/server/bootstrap.js:120:5)\n    at Command.<anonymous> (/usr/share/kibana/src/cli/serve/serve.js:216:5)"},"log":{"level":"FATAL","logger":"root"},"process":{"pid":7},"trace":{"id":"c4729231c8755337d67403e9e702f4ce"},"transaction":{"id":"50b0d300ed21358b"}}

**Reply by tqtran** [reply]
[kibana.log](https://github.com/Security-Onion-Solutions/securityonion/files/9343954/kibana.log)

**Reply by weslambert** [reply]
Are you still experiencing issues?

**Reply by tqtran** [reply]
hi Wes,
Thank you for the follow-up.

Kibana has been 404 since the update.  Since then, soup has updated SO to 2.3.160, but it still isn't resolving Kibana's 404 issue.  I've tried so-docker-refresh to refresh the container images, so I think this may be a configuration issue.

Is there a command to reset Kibana to defaults?  We haven't done any customization to Kibana since the installation.

Fortunately, the new SO Dashboards feature has provided us with an alternative way to view the data in the mean time.
